mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/util/generation/GeneratedCodePatterns.java

```diff
@@ -63,18 +70,48 @@
 
   /** Starts the generation of a method that overrides an interface method. */
   public static MethodSpec.Builder override(ExecutableElement interfaceMethod) {
+    return override(interfaceMethod, Collections.emptyMap());
+  }
+
+  public static MethodSpec.Builder override(
+      ExecutableElement interfaceMethod, Map<Name, TypeElement> typeParameters) {
     MethodSpec.Builder result =
         MethodSpec.methodBuilder(interfaceMethod.getSimpleName().toString())
             .addAnnotation(Override.class)
             .addModifiers(Modifier.PUBLIC)
-            .returns(ClassName.get(interfaceMethod.getReturnType()));
+            .returns(getTypeName(interfaceMethod.getReturnType(), typeParameters));
     for (VariableElement parameterElement : interfaceMethod.getParameters()) {
-      result.addParameter(
-          ClassName.get(parameterElement.asType()), parameterElement.getSimpleName().toString());
+      TypeName type = getTypeName(parameterElement.asType(), typeParameters);
+      result.addParameter(type, parameterElement.getSimpleName().toString());
     }
     return result;
   }
 
+  private static TypeName getTypeName(TypeMirror mirror, Map<Name, TypeElement> typeParameters) {
```

Comment:
Feels like there should be an easier way to do this, but was the best I could come up with.